packages/stream-model-instance/src/model-instance-document.ts

```diff
@@ -85,37 +61,20 @@ function headerFromMetadata(
   metadata: ModelInstanceDocumentMetadataArgs | StreamMetadata | undefined,
   genesis: boolean
 ): CommitHeader {
-  if (typeof metadata?.schema === 'string') {
-    try {
-      CommitID.fromString(metadata.schema)
-    } catch {
-      throw new Error('Schema must be a CommitID')
-    }
-  }
 
   const header: CommitHeader = {
     controllers: metadata?.controllers,
```

Comment:
```suggestion
  const header: CommitHeader = {
    controllers: metadata?.controllers,
    mode: metadata.model,
    unique: uint8arrays.toString(randomBytes(12), 'base64')
```
Actually, let's just do this.

Here I'm assuming that it's only possible to set metadata in the genesis record. Otherwise we need to put the `unique` property inside of the `genesis` condition.

Reply:
Checked and the `unique` property has to be set only for the genesis record, so moved it into the `genesis` condition block where it was originally.